### PR TITLE
Cluster support

### PR DESCRIPTION
**Pull request by leonhardhennig**
Note: I had to update some code / dependencies to get it working that will overlap / conflict with code in the branch update_dependencies and 8-allennlp-support. Once you reviewed, I will do the merge so that these branches have priority and only script changes from the cluster_support branch are merged.

### CODE REVIEW

File: scripts/cluster/wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+pip install .
+
+# This runs your wrapped command
+"$@"
```
Comment (GabrielKP) [inline scripts/cluster/wrapper.sh:6]: newlines at end of file?

File: scripts/cluster/ner.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  --save_steps 8500 \
+  --logging_steps 8500 \
+  --max_seq_length 128 \
+  --per_gpu_eval_batch_size=8 \
+  --per_gpu_train_batch_size=8 \
+  --learning_rate 2e-5 \
+  --num_train_epochs 5.0 \
+  --overwrite_cache \
+  --overwrite_output_dir \
+  --output_dir ./experiments/ner
```
Comment (GabrielKP) [inline scripts/cluster/ner.sh:21]: newline eof?

File: scripts/cluster/binary_relation_clf.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  --logging_steps 8500 \
+  --max_seq_length 128 \
+  --per_gpu_eval_batch_size=8 \
+  --per_gpu_train_batch_size=8 \
+  --learning_rate 2e-5 \
+  --num_train_epochs 5.0 \
+  --overwrite_cache \
+  --overwrite_output_dir \
+  --entity_handling mark_entity_append_ner \
+  --output_dir ./experiments/binary_relation_clf
```
Comment (GabrielKP) [inline scripts/cluster/binary_relation_clf.sh:22]: newline at eof?

File: scripts/cluster/readme.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+--container-mounts=/netscratch/$USER:/netscratch/$USER,/ds:/ds:ro,"`pwd`":"`pwd`" \
+--container-image=/netscratch/enroot/nvcr.io_nvidia_pytorch_21.07-py3.sqsh \
+--container-workdir="`pwd`" ./scripts/cluster/wrapper.sh pip list
+```
+
+### Train Relation Classifier
+Note: Maybe adapt the number gpus and number of tasks etc in scripts/cluster/batch.sh or on the command line
+
+```bash
+./scripts/cluster/batch.sh -p <PROFILE> --gpus=1 ./scripts/cluster/wrapper.sh ./scripts/cluster/binary_relation_clf.sh
```
Comment (GabrielKP) [inline scripts/cluster/readme.md:25]: Why is it called `<PROFILE>`?

File: scripts/cluster/wrapper.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (GabrielKP) [inline scripts/cluster/wrapper.sh:1]: Maybe it is cumbersome to have an own wrapper.sh if the only thing it does is pip install . (which could be easily included into the other scripts? Or maybe call the wrapper installer?

**Review by GabrielKP — APPROVED** [review]
Only minor things and nothing which is incompatible to changes in other branches

**Comment by GabrielKP** [comment]
Additionally, in my opinion all merge conflicts in `sherlock/documents.py` should take the incoming change from `cluster_support`.

File: scripts/cluster/readme.md (hunk 4 above)
Comment (leonhardhennig) [reply scripts/cluster/readme.md:25]: just a placeholder for your profile of choice . better without < > ?

File: scripts/cluster/wrapper.sh (hunk 5 above)
Comment (leonhardhennig) [reply scripts/cluster/wrapper.sh:1]: not sure either. usually we had 2 scripts - one for the srun command with predefined standard args, and one for the "set up" of the VM / conda env. so the usual call to a script was like "srun-script [additional-cl-args-for-slurm] setup-script python my-evaluation-script parameters". keeping batch.sh and wrapper/installer.sh separate allows to add args to batch.sh, before wrapper.sh is run. but I guess we could move "pip install ." to the task-specific scripts, but that would duplicate it :)

**Pull request merged**